### PR TITLE
tests: add mysql-cdc tests

### PR DESCRIPTION
**Pull request by nrainer-materialize**
### Issues
https://github.com/MaterializeInc/materialize/issues?q=is%3Aissue+author%3Anrainer-materialize+label%3AT-mysql+label%3AC-bug

Detect closed issues: https://buildkite.com/materialize/nightlies/builds/6559

### CODE REVIEW

**Comment by nrainer-materialize** [comment]
Already closed issues: https://buildkite.com/materialize/nightlies/builds/6510

File: test/mysql-cdc/two-publications-one-table.td [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+1
+
+$ mysql-execute name=mysql
+INSERT INTO t1 VALUES (2);
+
+> SELECT * FROM t1_1;
+1
+2
+
+> SELECT * FROM t1_1;
```
Comment (rjobanp) [inline test/mysql-cdc/two-publications-one-table.td:62]:
```suggestion
> SELECT * FROM t1_2;
```

File: test/mysql-cdc/truncate-table.td [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  FROM MYSQL CONNECTION mysql_conn
+  FOR ALL TABLES;
+
+> SELECT * FROM t1;
+1
+
+$ mysql-execute name=mysql
+TRUNCATE TABLE t1;
+
+> SELECT * FROM t1;
```
Comment (rjobanp) [inline test/mysql-cdc/truncate-table.td:47]: shouldn't this be an error?

File: test/mysql-cdc/alter-table-after-source.td [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+CREATE TABLE alter_drop_unique (f1 INTEGER UNIQUE);
+INSERT INTO alter_drop_unique VALUES (1);
+
+CREATE TABLE alter_add_unique (f1 INTEGER);
+INSERT INTO alter_add_unique VALUES (1);
+
+CREATE TABLE alter_extend_column (f1 VARCHAR(2));
+INSERT INTO alter_extend_column VALUES ('ab');
+
+# TODO: #25032 (decimal type)
```
Comment (rjobanp) [inline test/mysql-cdc/alter-table-after-source.td:70]: this should be possible on latest main

File: test/mysql-cdc/alter-table-after-source.td [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+ab
+
+$ mysql-execute name=mysql
+ALTER TABLE alter_extend_column MODIFY f1 VARCHAR(20);
+INSERT INTO alter_extend_column VALUES ('abcd');
+
+! SELECT * FROM alter_extend_column;
+contains:incompatible schema change
+
+# TODO: #25032 (decimal type)
```
Comment (rjobanp) [inline test/mysql-cdc/alter-table-after-source.td:283]: ditto

File: test/mysql-cdc/mysql-cdc.td [hunk 5]
```diff
@@ -0,0 +1,686 @@
...
+
+INSERT INTO pk_table VALUES (3, 'three');
+
+CREATE TABLE nonpk_table (f1 INTEGER, f2 INTEGER);
+INSERT INTO nonpk_table VALUES (1, 1), (1, 1);
+INSERT INTO nonpk_table VALUES (2, 2), (2, 2);
+
+# no range types in MySQL
+
+# TODO: #25032: add numeric_col DECIMAL(8,4) with 1234.5678 as value
```
Comment (rjobanp) [inline test/mysql-cdc/mysql-cdc.td:54]: possible on latest main

**Review by rjobanp — APPROVED** [review]
this is great!! I took a scan through and noted some fixes, though I'm not sure I got everything

File: test/mysql-cdc/two-publications-one-table.td (hunk 1 above)
Comment (nrainer-materialize) [reply test/mysql-cdc/two-publications-one-table.td:62]: Well spotted, thanks!

✅

File: test/mysql-cdc/truncate-table.td (hunk 2 above)
Comment (nrainer-materialize) [reply test/mysql-cdc/truncate-table.td:47]: True as well! Changed to `count(*)` expecting `0`, commented out, and linked #25051.

File: test/mysql-cdc/alter-table-after-source.td (hunk 3 above)
Comment (nrainer-materialize) [reply test/mysql-cdc/alter-table-after-source.td:70]: Thanks! (By the way, the closed-issue-detector job in nightly would have complained when the referenced issue is closed 😀.)

✅ Addressed.
I will add further test for decimal as follow-up.

**Pull request merged**